src/extras/curves/EllipseCurve.js

```diff
@@ -24,23 +24,41 @@ THREE.EllipseCurve.prototype.constructor = THREE.EllipseCurve;
 
 THREE.EllipseCurve.prototype.getPoint = function ( t ) {
 
-	var deltaAngle = this.aEndAngle - this.aStartAngle;
+	// Calculate deltaAngle just once.
+	var twoPi = Math.PI * 2;
+	if ( this.deltaAngle === undefined ) {
 
-	if ( deltaAngle < 0 ) deltaAngle += Math.PI * 2;
-	if ( deltaAngle > Math.PI * 2 ) deltaAngle -= Math.PI * 2;
+		var deltaAngle = this.aEndAngle - this.aStartAngle;
+		var samePoints = Math.abs( deltaAngle ) < Number.EPSILON;
 
-	var angle;
+		while ( deltaAngle < 0 ) deltaAngle += twoPi;
+		while ( deltaAngle > twoPi ) deltaAngle -= twoPi;
 
-	if ( this.aClockwise === true ) {
+		if ( deltaAngle < Number.EPSILON ) {
 
-		angle = this.aEndAngle + ( 1 - t ) * ( Math.PI * 2 - deltaAngle );
+			if ( samePoints ) {
 
-	} else {
+				deltaAngle = 0;
 
-		angle = this.aStartAngle + t * deltaAngle;
+			} else {
+
+				deltaAngle = twoPi;
+
+			}
+
+		}
+
+		if ( this.aClockwise === true && deltaAngle != twoPi && !samePoints ) {
+
+			deltaAngle = deltaAngle - twoPi;
+
+		}
+
+		this.deltaAngle = deltaAngle;
```

Comment:
I didn’t want to expose deltaAngle, I wanted to calculate it just once, and as I’m used to language where private property is the default, I forget that in JavaScript I would expose deltaAngle.
Anyway, as all the parameters of the ellipse are stored as member, as you point it, all are exposed and all can be modified. So deltaAngle should be recalculated for each call, no need to store it.

So the code inside the if should be kept (without the final `this.deltaAngle = deltaAngle;`) but without the if itself.
`var angle = this.aStartAngle + t * this.deltaAngle;`
becomes
`var angle = this.aStartAngle + t * deltaAngle;`